src/metabase/models/permissions_group.clj

```diff
@@ -100,12 +100,24 @@
     (when group-name
       (check-name-not-already-taken group-name))))
 
+(defn- grant-subscription-permission
```

Comment:
```suggestion
(defn- grant-subscription-permission!
```

https://guide.clojure.style/#naming-unsafe-functions

This function has side-effects. This wouldn't be safe in an STM transaction because if you ran it a second time it would fail because that entry would already exist. Group ID + `object` has a unique constraint.

If you're calling functions that end in `!` for side-effects is side of another function, that function's name should end in `!` as well.

Reply:
It also seems to me like **this** function belongs in `metabase.models.permissions` seeing how both `grant-permissions!` and other related similar functions all live there. You'd still have to require-resolve but it would only be one function to `resolve` instead of two